### DISCUSSION TITLE
Signature, expiration ,version get changed in all metadata files [except root.json and 1.root.json] when i add new bundle inside the targets

### DISCUSSION THREAD

**Discussion by ritishbhardwaj** [dennisvang/tufup#139] [Q&A]
Signature, expiration ,version get changed in all metadata files [except root.json and 1.root.json] when i add new bundle inside the targets using repo_add_bundle. Can i know why is it happening like if the signatures are getting modifying everytime, will this make any side effects while updating on the client side machine where this application is getting installed. [I know that on the client side only root.json is available!!!!]
And why not the version of root.json is getting changed!!!!.
Can i change the version of the root.json in near future?. if yes, then could you provide me reference from where i can know how to do it? And will it change the 1.root.json to 2.root.json or else?
Do i need to concern it before deploying it to the production?
<img width="909" alt="sig,expiration,version changing" src="https://github.com/dennisvang/tufup/assets/81488252/dc01c42c-c263-440f-8319-54f7a9888e40">

**Comment by dennisvang** [comment]
>Signature, expiration ,version get changed in all metadata files [except root.json and 1.root.json] when i add new bundle inside the targets using repo_add_bundle. Can i know why is it happening like if the signatures are getting modifying everytime, will this make any side effects while updating on the client side machine where this application is getting installed. ...

@ritishbhardwaj This is intentional: it is the way it is supposed to work. 

As you can see in your screenshot, the metadata object has a `signatures` member and a `signed` member. Every time something inside the `signed` member changes, that will require a new signature.

Also see #95

>... [I know that on the client side only root.json is available!!!!]

The client downloads the latest versions of root.json from your server, together with the other metadata files. See [detailed client workflow][1] in the TUF spec.

>And why not the version of root.json is getting changed!!!!. 

The root.json `signed` member has not changed, so, as long as it has not expired, it will *not* need to be re-signed, so the version will *not* change.

Please refer to the TUF spec where it is all [explained in great detail][2].

>Can i change the version of the root.json in near future?. if yes, then could you provide me reference from where i can know how to do it? 
And will it change the 1.root.json to 2.root.json or else?
Do i need to concern it before deploying it to the production?

As mentioned in the [TUF spec][3], root.json *"specifies trusted keys for the other top-level roles"*, so it only needs to be changed if you add, change, or remove trusted keys, *or* if the file has expired.

There's an example of a root key rotation in the repo workflow example file: 
https://github.com/dennisvang/tufup/blob/4bb16ad228b4b6f39e42527cc4367e501d728142/examples/repo/repo_workflow_example.py#L193-L205

Again, it is ***strongly*** advised to read the [TUF specification][2] to understand how this all works.

[1]: https://theupdateframework.github.io/specification/latest/#detailed-client-workflow
[2]: https://theupdateframework.github.io/specification/latest/#system-overview
[3]: https://theupdateframework.github.io/specification/latest/#metadata-files
